### DISCUSSION TITLE
SideroLink

### DISCUSSION THREAD

**Discussion by ryjogo** [siderolabs/talos#8756] [General]
in relation to https://github.com/siderolabs/talos/issues/8748#issuecomment-2117185547

Is SideroLink a standalone api that can be deployed for use with "aggregating" nodes? My use case is 1000's of nodes, that could connect to SideroLink. I could then perhaps tunnel any API calls through SideroLink

I couldn't find much documentation on this, but would this usecase fit?

**Comment by smira** [comment]
SideroLink can absolutely easily support thousands of nodes.

The repository https://github.com/siderolabs/siderolink provides a base "server" implementation with some stuff (specific to the usecase) provided as stubs.

If built from source, `siderolink-agent` provides a basic implementation Talos can connect to, and it already establishes Wireguard tunneling. 

The missing bits are mostly about validating incoming connections, persisting information about connected machines (to support restarts), and things like that. They can be plugged in via the library interfaces as needed.
